In [ ]:
library(tidyverse)
# Note that it can take some time to install the faraway package
install.packages('faraway')

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘minqa’, ‘nloptr’, ‘Rcpp’, ‘RcppEigen’, ‘lme4’




# STATS 306
## Homework 10: Linear Regression

For each problem, enter the R code in the cell marked "YOUR SOLUTION HERE".

## Problem 1: Linear regression on `Seatpos` Data (3 points)

For this problem, we will use the `seatpos` data in `faraway` package. You can find the R documentation for this data set here [https://www.rdocumentation.org/packages/faraway/versions/1.0.8/topics/seatpos].

In [ ]:
data(seatpos, package="faraway")

**1(a)** Given a new data set, it is a good idea to do data exploration first. Specifically, check the pairwise relationship among the variables in this data set. You can create a scatterplot for each pair of variables in `seatpos`. Describe any relationships among these variables. (1 point)

Hint: You may use `pairs()` function to generate a grid of scatterplots for each pair of variables in a data frame.

In [ ]:
# YOUR SOLUTION HERE

**1(b)** Fit a linear model using `hipcenter` as the response variable and all the remaining variables as predictors. Report the summary of the fitted model with interpretations. Does the summary show any sign of the patterns in 1(a)? (1 point)

In [ ]:
# YOUR SOLUTION HERE

**1(c)** In the previous questions, we notice there exsits the collinearity issue for some predictors. To overcome this issue, reduce the number of predictors and fit a new linear model. Report the summary of this new model with **comments**. (1 point)

Hint: When you choose a subset of predictors, you may consider their meanings and correlations with other predictors. There is no single answer for choosing the predictors.

In [ ]:
# YOUR SOLUTION HERE

## Problem 2: Linear regression on United Nation Data (2 points)

The data in the file UN11 from http://users.stat.umn.edu/~sandy/alr4ed/data/UN11.csv contains several variables, including ppgdp, the gross national product per person in U.S. dollars, and fertility, the birth rate per 1000 females, both from the year 2009. The data are for 199 localities, mostly UN member countries, but also other areas such as Hong Kong that are not independent countries. The data were collected from United Nations (2011). We will study the dependence of fertility on ppdgp.

In [ ]:
un11 = read.csv("http://users.stat.umn.edu/~sandy/alr4ed/data/UN11.csv")

**2(a)** Create a scatterplot of `fertility` versus `ppgdp`. Does a straight-line relationship seem to be suitable for this graph?  (0.5 points)

In [ ]:
# YOUR SOLUTION HERE

**2(b)** Based on the information shown in 2(a), if you think a linear relationship is not suitable for `fertility` and `ppgdp`, create a linear model using some transformation of `fertility` and/or `ppgdp`. Otherwise, create a linear model directly using `fertility` and `ppgdp`. Report the summary of your fitted model and **explain** why you choose or not choose the type of transformation. (1.5 points)

Hint: Based on the scatter plot in 2(a), what type of transformation would change the relationship between `fertility` and `ppgdp` to a linear trend?

In [ ]:
# YOUR SOLUTION HERE

# Problem 3: Linear regression on `gala` Data

For this problem, we will use the `gala` data in `faraway` package. You can find the R documentation for this data set [here](https://www.rdocumentation.org/packages/faraway/versions/1.0.8/topics/gala).

In [ ]:
data(gala, package = "faraway")

**(a)** To test whether a linear relationship exists between two variables, we can use the function `cor.test()` to test the null hypothesis $H_0:\rho=0$ where $\rho$ is the Pearson's correlation coefficient. We want to test if there is a linear relationship between
the number of endemic species (`Endemics`) and
the area of the adjacent island (`Adjacent`).

In [ ]:
cor.test(gala$Endemics, gala$Adjacent)


	Pearson's product-moment correlation

data:  gala$Endemics and gala$Adjacent
t = 0.43889, df = 28, p-value = 0.6641
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.2861320  0.4301187
sample estimates:
       cor 
0.08265803 


We can also test the correlation between two variables is zero using simple linear model. Fit a regression model and show you get the *exact* same p-value obtained by `cor.test()`. How can we infer the correlation coefficient $\rho = 0.0827$ from the regression result? *(1 point)*

In [ ]:
# YOUR SOLUTION HERE

**(b)** While Pearson's correlation only measures the strength of a linear correlation between two random variables, Spearman's rank correlation test looks for correlations between the ranks of two random variables.

Perform a test of Spearman's correlation using `cor.test()` (0.25 point)

In [ ]:
# YOUR SOLUTION HERE

Show that regressing the rank of `Endemics` on the rank of `Adjacent` gives you you the exact same p-values and estimates for as Spearman's test. (0.5 point)

*Hint*: you may use `rank()` to get the ranks of values in a vector.

In [ ]:
# YOUR SOLUTION HERE

**(c)** Fit a multiple linear regression model with `Endemics` as the response, and `Area`, `Elevation`, `Nearest`, `Scruz`, `Adjacent` as predictors. Which predictors are statistically significant? *(0.75 point)*

In [ ]:
# YOUR SOLUTION HERE

**(d)** Compute the correlation of the values of the residuals and the fitted values. Plot the values of the residuals against the fitted values. Please comment on what you have learned regarding the OLS assumption from this correlation and the plot. *(1.5 point)*

*Hint*: you can `residual()` and `fitted.values()` to get residuals and fitted values respectively.

In [ ]:
# YOUR SOLUTION HERE

**(e)** Quantile-quantile plot (QQ plot) draws the correlation between a given sample and the normal distribution. We can use QQ plot to check the normality. Generate a QQ-plot for residuals and comment on your finding. *(1 point)*

In [ ]:
# YOUR SOLUTION HERE